community/ndp/packstream-v1/src/main/java/org/neo4j/packstream/PackListItemType.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.packstream;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * PackStream list types provide a representative marker byte for each valid PackStream
+ * type, specifically for use with lists. PackStream's regular way of denoting type combines type
+ * detail with other information, such as scale. Therefore, there is no precise one-to-one
+ * mapping between type and marker byte values elsewhere.
+ *
+ * The values chosen more subtype marker bytes however do deliberately overlap with the main set
+ * of markers. This should aid clarity, reduce confusion and potentially allow sharing of code
+ * across both schemes.
+ *
+ * The full list of subtypes are:
+ *
+ *       Type -> Marker   (Code)
+ *   ----------------------------
+ *        ANY -> NULL     (C0)
+ *    BOOLEAN -> TRUE     (C3)
+ *    INTEGER -> INT_8    (C8)
+ *      FLOAT -> FLOAT_64 (C1)
+ *      BYTES -> BYTES_8  (CC)
+ *       TEXT -> TEXT_8   (D0)
+ *       LIST -> LIST_8   (D4)
+ *        MAP -> MAP_8    (D8)
+ *   ----------------------------
+ *     STRUCT -> STRUCT_8 (00-7F)
+ *
+ */
+public class PackListItemType
```

Comment:
Given our discussion earlier today about the distinction between the `primitives` (or whatever we decide to call them) of PackStream and the domain-level type system those are used to build, maybe it'd make sense to make something like class be the strict definition of that domain-level type system? 

It would be really nice to have it super explicit that this represents the *domain* type system (aka. the Neo4j Type System), rather than the PackStream primitives + structs. 

Somewhat related, it irks me a bit that we'd have a special type just to represent the inner type of a list. While the only place we represent types currently will be as innertype of Lists, it seems it'd be useful to have it as a general mechanism to represent types?